app/views/api/v2/analytics/user_analytics/_monthly_gender_table.html.erb

```diff
@@ -1,48 +1,137 @@
-<!-- the "all" table -->
-<table class="progress-table <%= data_table_name %> hypertension:all">
+<!-- don't load up the tables for totals and diabetes if diabetes is off -->
+<% if statistics.dig(:metadata, :is_diabetes_enabled) %>
+  <!-- the "all" table for both HTN & DM across genders -->
+  <table class="progress-table <%= data_table_name %> all">
+    <tr class="total">
+      <td class="row-label">All-time</td>
+      <td>
+        <%= zero_if_unavailable(statistics.dig(:all_time,
+                                               :grouped_by_date,
+                                               :total,
+                                               resource)) %>
+      </td>
+    </tr>
+    <% @user_analytics.monthly_period_list.each do |month_date| %>
+      <tr>
+        <td class="row-label no-padding"><%= month_date.strftime("%B") %></td>
+        <td>
+          <%= zero_if_unavailable(statistics.dig(:monthly,
+                                                 :grouped_by_date,
+                                                 :total,
+                                                 resource,
+                                                 month_date)) %>
+        </td>
+      </tr>
+    <% end %>
+  </table>
+
+  <!-- diabetes-only -->
+  <!-- the "all genders" table -->
+  <table class="progress-table <%= data_table_name %> diabetes:all" style="display: none">
+    <tr class="total">
+      <td class="row-label">All-time</td>
+      <td>
+        <%= zero_if_unavailable(statistics.dig(:all_time,
+                                               :grouped_by_gender,
+                                               :diabetes,
+                                               resource)
+                                    .values
+                                    .sum) %>
+      </td>
+    </tr>
+    <% @user_analytics.monthly_period_list.each do |month_date| %>
+      <tr>
+        <td class="row-label no-padding"><%= month_date.strftime("%B") %></td>
+        <td>
+          <%= zero_if_unavailable(statistics.dig(:monthly,
+                                                 :grouped_by_date,
+                                                 :diabetes,
+                                                 resource,
+                                                 month_date)) %>
+        </td>
+      </tr>
+    <% end %>
+  </table>
+<% end %>
+
+<% Patient::GENDERS.each do |gender| %>
+  <!-- gender-wise tables -->
+  <table class="progress-table <%= data_table_name %> diabetes:<%= gender %>" style="display: none">
+    <tr class="total">
+      <td class="row-label">All-time</td>
+      <td>
+        <%= zero_if_unavailable(statistics.dig(:all_time,
```

Comment:
All of these deeply nested statistics calls are hard to read and follow and also couple us to the exact structure of the data.

We have a presenter available already for all these views I believe. Lets use that and make the views way simpler and a bit less coupled:

Something like

```ruby
class UserAnalyticsPresenter

  def all_time_stats_grouped_by_gender(htn_or_dm, resource, gender)
  # this can handle the zero logic and the dig logic...and whatever else
...
  def monthly_grouped_by_gender(htn_or_dm, resource, gender)
...
```

Then in the views:

```
  <table class="progress-table <%= data_table_name %> diabetes:<%= gender %>" style="display: none">
    <tr class="total">
      <td class="row-label">All-time</td>
      <td>
             <%= user_analytics.monthly_grouped_by_gender(:hypertension, resource, gender) %>
       </td>
```

And so on down the line for all the views.

Reply:
I went back and forth on this. I'd wanted the statistics map to remain frozen and hence not require any further method abstractions. This "map" is contractual and should be a big deal if changed.

But I see readability concerns and I'll go ahead and stick them in methods.